base/logging.jl

```diff
@@ -474,7 +474,8 @@ function current_logstate()
 end
 
 # helper function to get the current logger, if enabled for the specified message type
-@noinline function current_logger_for_env(std_level::LogLevel, group, _module)
+@noinline function current_logger_for_env(std_level::LogLevel, group::Union{String,Symbol}, _module::Module)
+    @nospecialize  # block const-prop
```

Comment:
I don't think this actually blocks constant prop in inference (does it?) --- but it probably should!

Reply:
Hmm, maybe not reliably. But it it seems to help in some cases. E.g., I've used it several times on keyword arguments to prevent https://github.com/JuliaLang/julia/issues/38983#issuecomment-757143042.

There was definitely a big difference in inference timing here, but you're right that it doesn't always seem to prevent const-prop.

Reply:
I don't think these types are correct either. I'm pretty sure `::Any` was right

Reply:
The test failures definitely says we need at least `Nothing`.